controllers/dataplane/openstackdataplanedeployment_controller.go

```diff
@@ -216,7 +216,7 @@ func (r *OpenStackDataPlaneDeploymentReconciler) Reconcile(ctx context.Context,
 					for certKey := range service.Spec.TLSCerts {
 						result, err := deployment.EnsureTLSCerts(ctx, helper, &nodeSet,
 							nodeSet.Status.AllHostnames, nodeSet.Status.AllIPs, service, certKey)
-						if err != nil {
+						if (err != nil) || (*result != ctrl.Result{}) {
```

Comment:
I don't think we need this addition here.  If err is non-nil, we end up returning *result - which can be ctrl.Result{} or otherwise.

Reply:
There may be a situation when the nil is not a ... proper nil.[0] I don't think this is the case here. As it only happens if you abuse interfaces though and we shouldn't work around it. Rather stop it at the source.

[0] https://dev.to/arxeiss/in-go-nil-is-not-equal-to-nil-sometimes-jn8

Reply:
@vakwetu you're right, this change wasn't needed. removed it in the latest update